### DISCUSSION TITLE
RTC (LSE) loses time after power off

### DISCUSSION THREAD

**Discussion by epikao** [stm32duino/Arduino_Core_STM32#1642] [Libraries] [answered]
Hello

I use the bluepill with simple example code from stm32duino RTC library version 1.2.0 (see below).
When turning power off, the time is always lost even though a battery (2.8V) is connected to Vbat.

What am I doing wrong?

Thank you

```
#include <STM32RTC.h>
...
uint8_t houres = 0;
uint8_t minutes = 0;
uint8_t secondes = 0;
uint32_t subsecondes;
...
void setup() {
...
    rtc.setClockSource(STM32RTC::LSE_CLOCK);
    rtc.begin(false); // initialize RTC 24H format
...
}

void loop(){
...
    rtc.getTime(&houres, &minutes, &secondes, &subsecondes);
...
  Serial.printf("%02d:%02d:%02d\n", houres, minutes, secondes);
...
}
```

**Comment by ABOSTM** [comment]
HI @epikao,
I tested on my bluepill and it is working fine.
Note: I used a stabilized power supply (2.6V) to be sure there is no drop on vbat voltage.

In the SimpleRTC example, did you remove the following lines ?
Otherwise, after you power on bluepill, the setup() function will be (re)excuted and clock will be (re)initialized.
```
  // Set the time
//  rtc.setHours(hours);
//  rtc.setMinutes(minutes);
//  rtc.setSeconds(seconds);

  // Set the date
//  rtc.setWeekDay(weekDay);
//  rtc.setDay(day);
//  rtc.setMonth(month);
//  rtc.setYear(year);
```

**Reply by epikao** [reply]
> Note: I used a stabilized power supply (2.6V) to be sure there is no drop on vbat voltage.
> 
> In the SimpleRTC example, did you remove the following lines ?

why 2.6V? My bluepill is directly connected over the micro usb connector to my laptop.
I use exactly the code see above, no more lines. So it looks everything is correct... funny...

If I press and hold the reset button and then release it after about 30 seconds then the time is displayed correctly ( the time has not been reset and has also moved forward by 30 seconds.)

But when I unplug the USB and plug it back in, it doesn't work....(time starts at 00:00:00 )
So yes, could be something with "drop on vbat voltage"... but why? Is the bluepill circuit not correct?

Thank you

**Comment by bxparks** [comment]
For what it is worth, I tested this [examples/SimpleRTC](https://github.com/stm32duino/STM32RTC/tree/main/examples/SimpleRTC) code on my Blue Pill, with the clock source set to `STM32RTC::LSE_CLOCK`, and using the new STM32RTC v1.2.0 library, with STM32duino 2.2.0. And everything seems to work fine for me. Both the date and time are preserved across resets and power off. I use 2xNiMH batteries on the VBAT, outputting 2.52V.

**Comment by epikao** [comment, marked as answer]
Sorry for the inconvenience. It was my battery.
I used an old CR2032 battery. Although I measured a voltage of approx. 2.8V at Vbat (with a simple multimeter) and without power supply, this measurement is probably not meaningful. 
With a new battery it works now.
I think that if I had measured with an oscilloscope, short voltage interruptions would have been measurable...
